flake.nix

```diff
@@ -77,6 +77,16 @@
         };
 
         formatter = pkgs.alejandra;
+
+        devShells.kernel = pkgs.mkShell {
+          packages = [pkgs.ncurses pkgs.pkg-config];
+          inputsFrom = [pkgs.linux_latest];
+          shellHook = ''
+            export src=${pkgs.linux_latest.src}
```

Comment:
this is tied to the x86 mainline. Does it use the latest pinned kernel for a channel e.g. 23.05, 23.11, unstable. or does it always take the bleeding edge main?

Reply:
I think it uses the one from the project's flake.lock file

Reply:
So when we update the flake.lock file, it will change to the latest latest

Reply:
I think @mikatammi is right in that it uses https://github.com/tiiuae/ghaf/blob/main/flake.nix#L18 as pkgs and the exact version is based on flake.lock.

Reply:
> this is tied to the x86 mainline. 

It's actually available to all architectures - 
```
ghaf (8228000) [$?]
❯ nix develop .#devShells.
.#devShells.aarch64-linux  .#devShells.riscv64-linux  .#devShells.x86_64-linux
```
though those other than `x86_64` are currently not relevant for us in the context of `linux_latest` as we depend on the vendor BSP kernel versions for `aarch64` and `riscv64` targets.
To resolve this - we could develop target specific devShells with specific kernel source. In that case, I proposed @mikatammi that we'd refactor `devShells` to their own directory and import from there. To keep main level `flake.nix` clean.